### ISSUE TITLE
Search functionality

### ISSUE THREAD

**Issue by astridos2go** [gorilla-devs/ferium#189]
## The problem
It can be a pain to search for your mods on Modrinth/CurseForge in the broswer to just to get the ID to install using Ferium

## My solutions
- Adding a `ferium search <search_term>` command, to search Modrinth and/or CurseForge
- Ferium will automatically search for similar names when using `ferium add <mod_id>`, so if you type `ferium add dynamic-crosshair` it would come up with a message like ```Adding mod... The project/repository does not exist; Did you mean 'ferium add dynamiccrosshair'?```

**Comment by theRookieCoder** [comment]
I would prefer a UI interface that updates results as you type, and as usual there will be a option-based interface too.

**Comment by noahdotpy** [comment]
I think this search functionality would need to be added to ferinth and furse as there is not currently a binding for searching for projects in either afaik.

**Comment by khuongduy354** [comment]
So it's `ferium search` then pop up a TUI. As we type, it fetches both modrinth and curseforge, combines into a sorted list, and display to us? 

This seems like a big feature. Is it started yet? Im a beginner, but let me join if possible, I'll do my best.

**Comment by theRookieCoder** [comment]
1. As Noah mentioned, search APIs haven't been implemented in furse and ferinth (ferinth's search API is being worked on at gorilla-devs/ferinth#9)
2. As I mentioned in [a duplicate issue](https://github.com/gorilla-devs/ferium/issues/281#issuecomment-1480995842), this feature is low-priority at the moment

**Comment by BinToss** [comment]
> I would prefer a UI interface that updates results as you type, and as usual there will be a option-based interface too.

I don't think I've ever seen a CLI app have that feature. It might be impossible without an interactive mode to "take over" the console. A hacky solution would clear and rewrite the buffer on user input, but that's slow and isn't supported by all terminal emulators (e.g. xterm.js)
The closest thing is an argument completer (sometimes called Tab Completion) which could be implemented for shells that support it. WinGet's Search command is an example of an argument completer that returns web-fetched results. If you have PowerShell 7 or later, `winget search gi`, CTRL+SPACE would show a list like (see image below) while TAB would complete the current "word" with the first result from that list.
![image](https://github.com/user-attachments/assets/9c2805b8-08a3-4aef-9f85-475c27b90370)

Non-windows shells like Bash, Fish, Zsh, et cetera have their own APIs for argument completions, so it would have to be implemented for every shell this project's hypothetical argument completion chooses to support.

**Comment by theRookieCoder** [comment]
> so it would have to be implemented for every shell

I don't think so, I was thinking of tab completions too for things like selecting profiles or game versions. I recall finding a wip library for cross-shell dynamic completion for clap, if I can find it again I'll link to it.
